### PR TITLE
688 exception 3050

### PR DESCRIPTION
**Pull request by alishaevn**
# Story
validate the parameters given to the reports
- #688 

# Expected Behavior Before Changes
- you could pass an invalid parameter and the report would render

# Expected Behavior After Changes
- if you pass an invalid parameter, a 3050 exception error will be rendered

# Screenshots / Video

image

# Notes

### CODE REVIEW

File: app/models/sushi.rb [hunk 1]
```diff
@@ -112,6 +112,21 @@ def rescued_last_month_available(*args)
...
 
+  ##
+  # @param params [Hash, ActionController::Parameters]
+  # @param allowed_parameters [Array<String>]
+  #
+  # @return [String]
+  # @raise [Sushi::Error::UnrecognizedParameterError] when any unrecognized params are given.
+  module ParameterValidation
+    def validate_paramaters(params = {}, allowed_parameters: [])
+      filtered_params = params.reject { |key, _| ['action', 'controller', 'format'].include?(key) }
```
Comment (jeremyf) [inline app/models/sushi.rb:123]:
```ruby
given = params.keys.map(&:to_s) - ['action', 'controller', 'format']
return true if (given & allowed_parameters).length == given.length

 raise Sushi::Error::UnrecognizedParameterError.new(data: "The given parameter(s) are invalid: #{(given - allowed_parameters).join(', ')}.")
```

On the original line 124 you mapped the keys to_s; but if there's a mismatch, you don't remap the keys to string.  This handles that mapping ensuring you're comparing strings to strings..

**Review by jeremyf — CHANGES_REQUESTED** [review]

File: app/models/sushi.rb (hunk 1 above)
Comment (alishaevn) [reply app/models/sushi.rb:123]: good catch. I hadn't tested for the latter statement and it didn't cross my mind.

Comment (alishaevn) [reply app/models/sushi.rb:123]: also, is there an actual benefit of returning true instead of just returning?

Comment (jeremyf) [reply app/models/sushi.rb:123]: No significant benefit…just a slight "Hey this `validate` method said true" is kind of a nice response (given that Nil is false)

**Review by jeremyf — APPROVED** [review]

File: app/models/sushi.rb [hunk 2]
```diff
@@ -112,6 +112,23 @@ def rescued_last_month_available(*args)
     end
   end
 
+  ##
+  # This module provides a common interface for validating parameters.
+  #
+  # @param params [Hash, ActionController::Parameters]
+  # @param allowed_parameters [Array<String>] a list of query parameters AND report filters that are allowed.
```
Comment (alishaevn) [inline app/models/sushi.rb:119]: @jeremyf asking for your new review due to this change. it also required a change in the specs.

Comment (jeremyf) [reply app/models/sushi.rb:119]: It looks good.  The specs are good to update.

**Review by jeremyf — APPROVED** [review]

**Pull request merged**